File: src/main/java/com/github/guikeller/jettyrunner/conf/JettyRunnerConfigurationType.java
package com.github.guikeller.jettyrunner.conf;

import com.github.guikeller.jettyrunner.util.IconUtil;
import com.intellij.execution.configurations.ConfigurationFactory;
import com.intellij.execution.configurations.ConfigurationType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * Jetty Runner Configuration Type
 * @see com.intellij.execution.configurations.ConfigurationType
 * @author Gui Keller
 */
public class JettyRunnerConfigurationType implements ConfigurationType {

    public JettyRunnerConfigurationType(){
        super();
    }

    @Override
    public String getDisplayName() {
        return "Jetty Runner";
    }

    @Override
    public String getConfigurationTypeDescription() {
        return "IntelliJ IDEA Jetty Runner";
    }

    @Override
    public Icon getIcon() {
        return IconUtil.getInstance().getIcon();
    }

    @Override
    @NotNull
    public String getId() {
        return "JettyRunner-By-GuiKeller";
    }

    @Override
    public ConfigurationFactory[] getConfigurationFactories() {
        JettyRunnerConfigurationFactory factory = new JettyRunnerConfigurationFactory(this);
        return new ConfigurationFactory[]{factory};
    }
}

File: src/main/kotlin/com/github/guikeller/jettyrunner/conf/JettyRunnerConfigurationType.kt
package com.github.guikeller.jettyrunner.conf

import com.github.guikeller.jettyrunner.util.IconUtil
import com.intellij.execution.configurations.ConfigurationFactory
import com.intellij.execution.configurations.ConfigurationType
import javax.swing.Icon

/**
 * Jetty Runner Configuration Type
 * @see com.intellij.execution.configurations.ConfigurationType
 *
 * @author Guy Keller
 */
class JettyRunnerConfigurationType : ConfigurationType {
    override fun getDisplayName(): String {
        return "Jetty-Runner"
    }

    override fun getConfigurationTypeDescription(): String {
        return "Jetty-Runner"
    }

    override fun getIcon(): Icon {
        return IconUtil.getInstance().getIcon()
    }

    override fun getId(): String {
        return "Jetty-Runner"
    }

    override fun getConfigurationFactories(): Array<ConfigurationFactory> {
        val factory = JettyRunnerConfigurationFactory(this)
        return arrayOf(factory)
    }
}
